libnmstate/varlink/nmstate_varlink.py

```diff
@@ -0,0 +1,244 @@
+#
+# Copyright (c) 2020 Red Hat, Inc.
+#
+# This file is part of nmstate
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+#
+import os
+import libnmstate
+import logging
+import errno
+import io
+import json
+import libnmstate.error as libnmError
+
+
+try:
+    import varlink
+
+except ModuleNotFoundError:
+    raise libnmError.NmstateDependencyError("python3 varlink module not found")
+
+
+logger = logging.getLogger()
+logger.setLevel(logging.DEBUG)
+log_stringio = io.StringIO()
+handler = logging.StreamHandler(log_stringio)
+handler.setLevel(logging.DEBUG)
+handler.setFormatter(
+    logging.Formatter(
+        '{"time": "%(asctime)s",'
+        + '"level": "%(levelname)s", "message": "%(message)s"}',
+        datefmt="%Y-%m-%d %H:%M:%S",
+    )
+)
+logger.addHandler(handler)
+
+service = varlink.Service(
```

Comment:
This should be created by `start_varlink_server()` or other functions instead of shared at module level.

The same reason for the race problem of logger.